staging/src/k8s.io/apiserver/pkg/util/webhook/webhook.go

```diff
@@ -62,7 +62,7 @@ type GenericWebhook struct {
 // Otherwise it returns false for an immediate fail.
 func DefaultShouldRetry(err error) bool {
 	// these errors indicate a transient error that should be retried.
-	if utilnet.IsConnectionReset(err) || apierrors.IsInternalError(err) || apierrors.IsTimeout(err) || apierrors.IsTooManyRequests(err) {
+	if utilnet.IsConnectionReset(err) || utilnet.IsConnectionLost(err) || apierrors.IsInternalError(err) || apierrors.IsTimeout(err) || apierrors.IsTooManyRequests(err) {
```

Comment:
The only retryable operations should be those that are idempotent or those that the error indicate that there is no risk to retry, IsConnectionLost means that the ping frame was not answered and is a connection level error that impacts all in-flight requests
https://cs.opensource.google/go/x/net/+/master:http2/transport.go

Reply:
I am not too sure about if it's idempotent but I do see some similarity between IsConnectionReset and IsConnectionLost.

Anyway, is it safe to make it retryable just for webhook authentication since it's idempotent (ie. review token)? cc @liggitt

Reply:
Agree with IsConnectionReset, since this can be sent after the server processed some data.
Jordan know better the mechanics of the webhooks, if all operations are idempotent then +1 to retry, otherwise we should alos remove the IsConnectionReset IMHO

Reply:
authn and authz webhooks are safe to retry

admission webhooks are generally safe to retry, are already retried at a higher level when there's an etcd write conflict, and provide a uid in the review body so a webhook that cared about idempotence (because it had side effects it wanted to minimize) could detect a retry

Reply:
IIUC, only authn & authz use `DefaultShouldRetry` https://sourcegraph.com/github.com/kubernetes/kubernetes@b790933e8aefdac194ba3c199b5e552a94107950/-/blob/staging/src/k8s.io/apiserver/pkg/util/webhook/webhook.go?L63%3A6=#tab=references